File: R/examples/basics.R

library(ddf)
dm <- DDFManager("spark")

write.table(mtcars, "/tmp/mtcars", row.names=F, col.names=F)

sql(dm, 'set hive.metastore.warehouse.dir=/tmp/hive/warehouse')
sql(dm, "drop table if exists mtcars")
sql(dm, "CREATE TABLE mtcars (mpg double, cyl int, disp double, hp int, drat double, wt double, qesc double, vs int, am int, gear int, carb string) ROW FORMAT DELIMITED FIELDS TERMINATED BY ' '")
sql(dm, paste0("LOAD DATA LOCAL INPATH '/tmp/mtcars' INTO TABLE mtcars"))

ddf <- sql2ddf(dm, "select * from mtcars")

colnames(ddf)

ncol(ddf)
nrow(ddf)

summary(ddf)

head(ddf)

daggr(mpg ~ vs + carb, ddf, FUN=mean)
daggr(ddf, agg.cols="sum(mpg), min(hp)", by="vs, am")

fivenum(ddf)

sample(ddf, 10L)

shutdown(dm)

File: python/examples/basics.py
from __future__ import unicode_literals
from ddf import DDFManager, DDF_HOME

dm = DDFManager('spark')

dm.sql('set hive.metastore.warehouse.dir=/tmp/hive/warehouse', False)
dm.sql('drop table if exists mtcars', False)
dm.sql("CREATE TABLE mtcars (mpg double, cyl int, disp double, hp int, drat double, wt double,"
       " qesc double, vs int, am int, gear int, carb string) ROW FORMAT DELIMITED FIELDS TERMINATED BY ' '", False)
dm.sql("LOAD DATA LOCAL INPATH '" + DDF_HOME + "/resources/test/mtcars' INTO TABLE mtcars", False)

ddf = dm.sql2ddf('select * from mtcars', False)

print('Columns: ' + ', '.join(ddf.colnames))

print('Number of columns: {}'.format(ddf.cols))
print('Number of rows: {}'.format(ddf.rows))

print(ddf.summary())

print(ddf.head(2))

print(ddf.aggregate(['sum(mpg)', 'min(hp)'], ['vs', 'am']))

print(ddf.five_nums())

print(ddf.sample(3))

dm.shutdown()

